GTC-3143 Titiler endpoint for new drivers of TCL

danscales opened the pull request:
GTC-3143 Titiler endpoint for new drivers of TCL

The new drivers dataset (wri_google_tree_cover_loss_drivers) has 7 categories of drivers. We are filtering by
umd_tree_cover_density_2000/v1.8.

Slight changes to update the arg descriptions in gross_emissions and net_flux.

Tested locally.

danscales committed:
GTC-3143 Titiler endpoint for new drivers of TCL

The new drivers dataset (wri_google_tree_cover_loss_drivers) has 7
categories of drivers. We are filtering by
umd_tree_cover_density_2000/v1.8.

Slight changes to update descriptions in gross_emissions and net_flux.

Tested locally.

dmannarino commented on app/routes/titiler/algorithms/tree_cover_loss_drivers.py:
```diff
@@ -0,0 +1,103 @@
+from rio_tiler.models import ImageData
+from titiler.core.algorithm import BaseAlgorithm
+
+Colors: namedtuple = namedtuple("Colors", ["red", "green", "blue"])
```
Probably time we factored this out?

dmannarino commented on app/routes/titiler/algorithms/tree_cover_loss_drivers.py:
```diff
@@ -0,0 +1,103 @@
+            4: Colors(81, 164, 78),    # Forest management
+            5: Colors(137, 81, 40),    # Wildfires
+            6: Colors(163, 84, 160),   # Settlements and infrastructure
+            7: Colors(58, 32, 154),    # Other natural disturbances
```
Thanks for labeling!

dmannarino replied:
I almost wonder if you should put them in a data structure (named tuple?), rather than in just comments? Just an idea.

danscales replied:
I don't really see how named tuple will help, since that gives names to the fields of a tuple.   Also, the value of the key (1-7) is important, since that's the actual encoding of course.  So, I think the main thing would be to make the value into a tuple as  [ Colors(81, 164, 78), "Forest management].  Then when I use the conf_colors, I would alway just use the Colors part of the tuple in  create_true_color_rgb.  But that doesn't seem to add any extra documentation value over having the comment there. So, for now, I'll leave as is.

dmannarino commented on app/routes/titiler/wri_tree_cover_loss_drivers.py:
```diff
@@ -0,0 +1,83 @@
+from .algorithms.tree_cover_loss_drivers import TreeCoverLossDrivers
+from .readers import AlertsReader
+
+DATA_LAKE_BUCKET = os.environ.get("DATA_LAKE_BUCKET")
```
Probably should move to settings/globals.py?

danscales replied:
Yes, we can factor things out as you suggest once all the titiler endpoints have settled down a bit. (We haven't even seen Maanas' TCL endpoint yet.)

dmannarino reviewed (approved):
Just a few non-blocking suggestions.

solomon-negusse commented on app/routes/titiler/algorithms/tree_cover_loss_drivers.py:
```diff
@@ -0,0 +1,103 @@
+    def create_true_color_rgb(self):
+        r, g, b = self._rgb_zeros_array()
+
+        for k, colors in self.conf_colors.items():
```
Minor but equality check may make it clearer that this is categorical data.

danscales replied:
Done!

solomon-negusse reviewed (approved):
LOoks good.. just left one comment

danscales committed:
Change to equality test in create_true_color_rgb()

Change to equality test in create_true_color_rgb()
to make it clearer that this is a categorical data
